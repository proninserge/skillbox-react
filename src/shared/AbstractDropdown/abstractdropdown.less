.container {
    width: 100%;
    flex: 1 1 0%;
}

.listContainer {
    position: relative;
}

.list {
    position: absolute;
    top: 0;
    right: 0;
    width: 100%;
    height: 100%;
}
