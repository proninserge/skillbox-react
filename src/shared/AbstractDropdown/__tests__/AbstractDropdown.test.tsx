import {shallow, configure} from 'enzyme';
import {Dropdown} from '../AbstractDropdown';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('AbstractDropdown', () => {
    const wrapper = shallow(<Dropdown children={<div className="dropdown"/>} button={<button className="buttonToClick"/>} />);

    test('should render', () => {
        expect(wrapper).toBeDefined(); // check if a component is rendered
        expect(wrapper.find('div.container').isEmptyRender()).toBeFalsy();
    });

    test ('should render snapshot', () => {
        const wrapperForSnapshot = shallow(<Dropdown children={<div />} button={<button />} />);
        expect(wrapperForSnapshot).toMatchSnapshot();
    });

    test ('should evoke useState and change state when clicked on the closest to the button div', () => {
        const setIsDropdownOpen = jest.fn();
        const handleState = jest.spyOn(React, 'useState');
        const handleOpen: any = (isDropdownOpen: any) => [isDropdownOpen, setIsDropdownOpen];
        
        handleState.mockImplementation(handleOpen);
        
        const wrapperForClick = shallow(<Dropdown children={<div className="dropdown"/>} button={<button className="buttonToClick" onClick={setIsDropdownOpen} />} />);
 
        const menuButton = wrapperForClick.find('button.buttonToClick');
        menuButton.closest('div').simulate('click');
        expect(setIsDropdownOpen).toBeTruthy();
    });

    test ('should render listContainer if isOpen has true', () => {
        const wrapperWithListContainer = shallow(<Dropdown isOpen={true} children={<div />} button={<button />} />);
        expect(wrapperWithListContainer.find('div.listContainer').isEmptyRender()).toBeFalsy();
    });

});