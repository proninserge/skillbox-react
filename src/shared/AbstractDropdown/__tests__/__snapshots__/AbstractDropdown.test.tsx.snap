// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbstractDropdown should render snapshot 1`] = `
<div
  className="container"
>
  <div
    onClick={[Function]}
  >
    <button />
  </div>
</div>
`;
